### ISSUE TITLE
[Bug Report] download manager reports 'socket connection closed'

### ISSUE THREAD

**Issue by ch604** [hkalexling/Mango#190]
**Describe the bug**
when opening the download manager, a pink popup arrives which says "Socket connection closed"

**To Reproduce**
1. begin a download via plugin (manganelo is working at the moment)
2. go to the download manager

**Expected behavior**
no errors

**Environment (please complete the following information):**
 - OS: debian 10 on raspberry pi 4
 - Browser: firefox latest
 - Mango Version: v0.21 on arm64v8 via docker

**Docker (if you are running Mango in a Docker container)**
default docker-compose.yml file is in use, save arm modification

**Additional context**
this doesnt seem to be detrimental to the operation of the downloader, unless it is supposed to be dynamically updating. clicking 'refresh' and 'clear downloaded' and all other functions work well, it only displays the error when loading the downloader for the first time.

**Comment by hkalexling** [comment]
Hi, thanks for the bug report!

Yes, it is supposed to be dynamically updating. A web socket connection is created on page load to update the download list every 5 seconds. Apparently the connection was closed for some reason. Here's the relevant JS code https://github.com/hkalexling/Mango/blob/8829d2e237889d0ec232a92a485b3ef4086ecaa0/public/js/download-manager.js#L9-L27

I couldn't reproduce this on my test server. I will try again later with an ARM machine and see if that's the culprit. Do you see any error messages in the browser console?

**Comment by ch604** [comment]
So I loaded up mango through its internal port/ip combination, and did it not replicate the error. The console on the page, though, shows this:

`WebSocket connection to 'wss://10.10.0.18:9000/api/admin/mangadex/queue' failed`

...where the mango connection through the proxied domain name shows both wss:// and ws:// failing.

I was able to narrow this down to a problem with my nginx proxy not being set up properly to forward web sockets. I was missing these two lines:

```
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
```

Once I added these to nginx, the socket started working correctly and all is well again :)
